google/cloud/storage/object_metadata.cc

```diff
@@ -20,38 +20,111 @@ namespace google {
 namespace cloud {
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
-ObjectMetadata ObjectMetadata::ParseFromJson(std::string const& payload) {
-  auto json = storage::internal::nl::json::parse(payload);
+ObjectMetadata ObjectMetadata::ParseFromJson(internal::nl::json const& json) {
   ObjectMetadata result{};
-  static_cast<CommonMetadata&>(result) =
-      internal::MetadataParser::ParseCommonMetadata(json);
+  static_cast<CommonMetadata<ObjectMetadata>&>(result) =
+      CommonMetadata<ObjectMetadata>::ParseFromJson(json);
+
+  if (json.count("acl") != 0) {
+    for (auto const& kv : json["acl"].items()) {
+      result.acl_.emplace_back(ObjectAccessControl::ParseFromJson(kv.value()));
+    }
+  }
+
   result.bucket_ = json.value("bucket", "");
-  result.generation_ =
-      internal::MetadataParser::ParseLongField(json, "generation");
+  result.cache_control_ = json.value("cacheControl", "");
+  result.component_count_ = internal::ParseLongField(json, "componentCount");
+  result.content_disposition_ = json.value("contentDisposition", "");
+  result.content_encoding_ = json.value("contentEncoding", "");
+  result.content_language_ = json.value("contentLanguage", "");
+  result.content_type_ = json.value("contentType", "");
+  result.crc32c_ = json.value("crc32c", "");
+  if (0 != json.count("customerEncryption")) {
```

Comment:
Nit: Be consistent with which side of the `!=` operator you place the literal value (here, it's written as `0 != json.count(...)`, but above, it's written as `json.count(...) != 0`).

Reply:
Fixed.